### PR TITLE
[Issue-2894] fix execution of single binary path

### PR DESCRIPTION
**Pull request by whizzzkid**
The issue is described in detail: #2894

### CODE REVIEW

**Comment by wraithgar** [comment]
Thank you for submitting this PR.  Could you add a test for this behavior?

**Comment by isaacs** [comment]
This should use the `npm-normalize-package-bin` module (which we use when parsing package.json files) instead of just an inline affordance for one situation.

Alternatively: it might be a good idea to have pacote.manifest always call `npm-normalize-package-bin` on the manifests it returns, and then this would never be an issue.  (Protection in depth is potentially still valuable, of course.)

**Comment by isaacs** [comment]
Hm, actually, I think this should be fixed in `npm-pick-manifest`, since it's the only place where we get a bin that might not already be normalized.  Will have a patch for that soon.

**Comment by whizzzkid** [comment]
@isaacs I implemented the change using `npm-normalize-package-bin` I didn't even know something like this existed (just npm things I guess 😛 )

Let me know if you want me to fix `npm-pick-manifest` instead.

File: package-lock.json [hunk 1]
```diff
@@ -14500,8 +14502,8 @@
     },
     "npm-normalize-package-bin": {
       "version": "1.0.1",
-      "resolved": "https://registry.npmjs.org/npm-normalize-package-bin/-/npm-normalize-package-bin-1.0.1.tgz",
-      "integrity": "sha512-EPfafl6JL5/rU+ot6P3gRSCpPDW5VmIzX959Ob1+ySFUuuYHWHekXpwdUZcKP5C+DS4GEtdJluwBjnsNDl+fSA=="
+      "resolved": "https://artifactory.appdirect.tools/artifactory/api/npm/npm-repo/npm-normalize-package-bin/-/npm-normalize-package-bin-1.0.1.tgz",
```
Comment (ruyadorno) [inline package-lock.json:14505]: oops

**Comment by ruyadorno** [comment]
closing in favor of https://github.com/npm/npm-pick-manifest/pull/32

thank you @whizzzkid! 🎉

**Comment by whizzzkid** [comment]
@ruyadorno I think we should still have a unit test for this scenario, I'll create one!

**Pull request closed without merging**